files/image_config/hostcfgd/hostcfgd

```diff
@@ -219,6 +220,537 @@ class AaaCfg(object):
+
+        mgmt_intf = self.config_db.get_entry(CFG_MGMT_INTF_TABLE, ifname)
+        if mgmt_intf == {}:
+            mgmt_intf['ipv4_dhcp_client'] = "true"
+            mgmt_intf['ipv6_dhcp_client'] = "true"
+            self.config_db.set_entry(CFG_MGMT_INTF_TABLE, ifname, mgmt_intf)
+            self.app_db.set_entry(APP_MGMT_INTF_TABLE, ifname, mgmt_intf)
+
+    def mgmt_intf_mtu_set (self, ifname, mtu, appIntf):
+        cmd = 'ifconfig {} mtu {}'.format(ifname, mtu)
+        syslog.syslog(syslog.LOG_INFO, "Configure MTU, cmd - {}".format(cmd))
+        try:
+            subprocess.check_call(cmd, shell=True)
+            appIntf['mtu'] = mtu
+        except subprocess.CalledProcessError as err:
+            syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}"
+                    .format(err.cmd, err.returncode, err.output))
+
+    def mgmt_intf_admin_status_set (self, ifname, admin_status, appIntf):
+        cmd = 'ifconfig {} {}'.format(ifname, admin_status)
+        syslog.syslog(syslog.LOG_INFO, "Configure admin_status, cmd - {}".format(cmd))
+        try:
+            subprocess.check_call(cmd, shell=True)
+            appIntf['admin_status'] = admin_status
+        except subprocess.CalledProcessError as err:
+            syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}"
+                    .format(err.cmd, err.returncode, err.output))
+
+    def mgmt_intf_description_set (self, ifname, description, appIntf):
+        syslog.syslog(syslog.LOG_INFO, "Configure description, cmd - {}".format(description))
+        appIntf['description'] = description
+
+    def mgmt_intf_speed_set (self, ifname, speed, autoneg, appIntf):
+        duplex = "full"
+        cmd = 'ethtool -s {} speed {} duplex {} autoneg {}; ifup {}'.format(ifname, speed, duplex, autoneg, ifname)
+        syslog.syslog(syslog.LOG_INFO, "Configure SPEED, cmd - {}".format(cmd))
+
+        try:
+            subprocess.check_call(cmd, shell=True)
+            appIntf['speed'] = speed
+        except subprocess.CalledProcessError as err:
+            syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}"
+                    .format(err.cmd, err.returncode, err.output))
+
+    def mgmt_intf_autoneg_set (self, ifname, autoneg, appIntf):
+        an = "on"
+        if autoneg == "true":
+            an = "on"
+        elif autoneg == "false":
+            an = "off"
+
+        cmd = "ethtool -s {} autoneg {}".format(ifname, an)
+        syslog.syslog(syslog.LOG_INFO, "Configure autoneg, cmd - {}".format(cmd))
+
+        try:
+            subprocess.check_call(cmd, shell=True)
+            appIntf['autoneg'] = autoneg
+        except subprocess.CalledProcessError as err:
+             syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(err.cmd, err.returncode, err.output))
+
+    def mgmt_intf_gwaddr_set(self, ifname, ip_prefix, gwaddr, op):
```

Comment:
I don't get the full use-case of this function to create ip prefix subnet to a gwaddr. May be this can be handled later separately.

Reply:
addressed